### DISCUSSION TITLE
setAppWidgetSize() and getAppWidgetSize() not working in BIP 5

### DISCUSSION THREAD

**Discussion by amirrouhandeh** [zepp-health/zeppos-forum#266] [Developer Feedback]
Already designed a custom app-widget and also tested the calorie template on Bip 5 . **The problem is the shortcut occupies the most space on the screen in a rectangular shape without any round corners! (tested on both watch and simulator)**

![image](https://github.com/zepp-health/zeppos-forum/assets/3074871/20a33856-d57c-46d0-810e-3305f7e86533)
![AppWidget-bug](https://github.com/zepp-health/zeppos-forum/assets/3074871/fb9deacf-85a0-448a-aa13-3c3545b4cb9c)
![AppWidget-bug2](https://github.com/zepp-health/zeppos-forum/assets/3074871/f258fac8-eecd-4e77-831a-41b8dc990ae3)

**Comment by MyoungXUE** [comment]
Hello, this is a known issue and we will fix it in the next OTA. In the current version, it is recommended to use `setTimeout` to delay some calls during the build lifecycle. This is the code snippet I provided to you.

At the same time, we will fix the documentation and examples as soon as possible. Thank you for your feedback.

```js
import { setAppWidgetSize, getAppWidgetSize } from '@zos/ui';

AppWidget({
  build(){
    setTimeout(() => {
      console.log('app-widget-size', getAppWidgetSize())
    }, 100)
  },
})
```

**Reply by amirrouhandeh** [reply]
thanks for your response, but it seems that we've discovered another bug!
setTimeout() is also not working on Bip 5!

```
setTimeout(() => {
  console.log('Hello Zepp OS')
}, 1000)
```
even this is not working!

**Reply by MyoungXUE** [reply]
Sorry my fault, not `build`, it is `onResume`

```js
import { setAppWidgetSize, getAppWidgetSize } from '@zos/ui';

AppWidget({
  onResume(){
    setTimeout(() => {
      console.log('app-widget-size', getAppWidgetSize())
    }, 10)
  },
})
```

**Reply by amirrouhandeh** [reply]
Now it works thanks to you and the height issue is fixed but the rounded corners still are not available.
I think it's because of the wrong margin value in Bip 5 SDK which is 65505!
![image](https://github.com/zepp-health/zeppos-forum/assets/3074871/3bb4a205-cd9e-4885-acd1-3a70b6137793)
![image](https://github.com/zepp-health/zeppos-forum/assets/3074871/c4f3266e-fa4d-4faa-b771-d224000473e3)

**Reply by MyoungXUE** [reply]
Hello, after confirmation, this is an emulator bug, and there is no such problem on the real machine, so don't worry about rounded corners. We will fix emumator ASAP.

**Reply by amirrouhandeh** [reply]
![SGCAM_20231114_0346147310.jpg](https://github.com/zepp-health/zeppos-forum/assets/3074871/ebba3112-7ca5-4799-a680-051efb596653)

So does it need an OTA update to get fixed? since it's not shown properly on the watch too!

**Reply by MyoungXUE** [reply]
I will confirm again and this issue will be fixed in the next OTA.
